packages/utils/src/ip-port-to-multiaddr.ts

```diff
@@ -27,21 +27,15 @@ export function ipPortToMultiaddr (ip: string, port: number | string): Multiaddr
     throw new CodeError(`invalid port provided: ${port}`, Errors.ERR_INVALID_PORT_PARAMETER)
   }
 
-  try {
-    // Test valid IPv4
-    new Address4(ip) // eslint-disable-line no-new
+  if (isIPv4(ip)) {
     return multiaddr(`/ip4/${ip}/tcp/${port}`)
-  } catch {}
-
-  try {
-    // Test valid IPv6
-    const ip6 = new Address6(ip)
-    return ip6.is4()
-      ? multiaddr(`/ip4/${ip6.to4().correctForm()}/tcp/${port}`)
-      : multiaddr(`/ip6/${ip}/tcp/${port}`)
-  } catch (err) {
-    const errMsg = `invalid ip:port for creating a multiaddr: ${ip}:${port}`
-    log.error(errMsg)
-    throw new CodeError(errMsg, Errors.ERR_INVALID_IP)
   }
+
+  if (isIPv6(ip)) {
```

Comment:
`@chainsafe/is-ip` doesn't support detecting [IPv4 in IPv6 addresses](https://www.ibm.com/docs/en/ts4500-tape-library?topic=functionality-ipv4-ipv6-address-formats) so we lose support for returning an `IPv4` address in this case.

Node.js recognises these sorts of addresses as `IPv6` so I don't know if this is a problem or not.

E.g.:


```console
% node
Welcome to Node.js v18.14.0.
Type ".help" for more information.
> const net = require('net')
undefined
> net.isIP('0:0:0:0:0:0:101.45.75.219')
6
```